patches/p1.js

```diff
@@ -0,0 +1,36 @@
+const Enmap = require("enmap");
+
+const v1Roles = new Enmap({name :'rolesData'});
+const v2Roles = new Enmap({name :'rolesDataV2'});
+/**
+ * Migration for the roles command's schema
+ */
+function migrateRolesToV2() {
+  console.log(`rolesV1-pre: ${v1Roles.export()}`);
+  console.log(`rolesV2-pre: ${v2Roles.export()}`);
+
+  //v1 formatted as guildId -> arr[roleNames]
+  //v2 is a documentStore of arbitraryId -> {roleInfo} e.g. name, guildId, etc. 
+  v1Roles.keyArray().forEach(guildId => {
+    let roleNames = v1Roles.ensure(guildId, []);
+    roleNames.forEach(roleName => {
+      v2Roles.set(guildId + ":" + roleName, {
```

Comment:
is guildId necessary here since its already keyed by guildId? Or is this just extra redundancy? 

EDIT: nvm, misread. how come you want to flatten it instead of keeping it separated by guildId?

Reply:
It makes it easier to do lookups with enmap's filter command.